pisi/cli/check.py

```diff
@@ -135,6 +138,24 @@ def run(self):
                 # Package is not installed
                 ctx.ui.info(_("Package %s not installed") % pkg)
 
+        # Filter broken packages that are not in a repository
+        pkgdb = pisi.db.packagedb.PackageDB()
+        custom_pkgs = [pkg for pkg in broken_pkgs if not pkgdb.has_package(pkg)]
+        broken_pkgs = [pkg for pkg in broken_pkgs if pkgdb.has_package(pkg)]
+
+        if custom_pkgs:
+            ctx.ui.warning(_("These packages are broken but are not found in a repository, they must be reinstalled manually:"))
+            ctx.ui.info(util.format_by_columns(sorted(custom_pkgs)))
+
+        if broken_pkgs:
+            ctx.ui.info("")
+            ctx.ui.warning(_("The following packages are broken:"))
+            ctx.ui.info(util.format_by_columns(sorted(broken_pkgs)))
+            if ctx.ui.confirm(_("Reinstall the broken packages?")):
+                pisi.api.install(broken_pkgs, reinstall=True)
```

Comment:
The underlying operation can throw an error, so we might want to catch that. I'm not sure that we can really meaningfully *do anything* about it, except just print a message so we don't just end up with a stack trace.